itest/lnd_send_coins_utxo_test.go

```diff
@@ -0,0 +1,200 @@
+			amt:                250_000,
+			sendCoinShouldFail: true,
+		},
+		{
+			name: "sendCoins amount specfied greater than " +
+				"available amount",
+			amt:                1_000_000,
+			sendCoinShouldFail: true,
+		},
+		{
+			name: "sendCoins with selected utxos, amt specified " +
+				"greater that select utxos funds",
+			selectedCoins: []btcutil.Amount{
+				50_000,
+				1_000,
+			},
+			amt:                200_000,
+			sendCoinShouldFail: true,
+		},
+	}
+
+	for _, tc := range tcs {
+		ht.Run(tc.name, func(t *testing.T) {
+			// Fund alice and compute the expected wallet balance.
+			totalinitialCoinSum := 0
+			for _, initialCoin := range initialCoins {
+				ht.FundCoins(initialCoin, alice)
+				totalinitialCoinSum += int(initialCoin)
+			}
+
+			ht.AssertWalletAccountBalance(
+				alice, lnwallet.DefaultAccountName,
+				int64(totalinitialCoinSum), 0,
+			)
+
+			defer func() {
+				// Fund additional coins to sweep in case the
+				// wallet contains dust.
+				ht.FundCoins(100_000, alice)
+
+				// Remove all funds from Alice.
+				sweepNodeWalletAndAssert(ht, alice)
+			}()
+
+			// Create an outpoint lookup for each unique amount.
+			lookup := make(map[int64]*lnrpc.OutPoint)
+			res := alice.RPC.ListUnspent(
+				&walletrpc.ListUnspentRequest{},
+			)
+			for _, utxo := range res.Utxos {
+				lookup[utxo.AmountSat] = utxo.Outpoint
+			}
+
+			// Map the selected coin to the respective outpoint.
+			var selectedOutpoints []*lnrpc.OutPoint
+			selectedOpLookup := make(map[string]struct{})
+			for _, selectedCoin := range tc.selectedCoins {
+				if o, ok := lookup[int64(selectedCoin)]; ok {
+					selectedOpLookup[o.TxidStr] = struct{}{}
+					selectedOutpoints = append(
+						selectedOutpoints, o,
+					)
+				}
+			}
+
+			// Generate address for the request.
+			p2wkhAddr, err := btcutil.NewAddressWitnessPubKeyHash(
+				make([]byte, 20), harnessNetParams,
+			)
+
+			require.NoError(t, err, "error generating "+
+				"target address for request")
+
+			sendCoinReq := &lnrpc.SendCoinsRequest{
+				Addr:       p2wkhAddr.String(),
+				SendAll:    tc.sendAll,
+				Outpoints:  selectedOutpoints,
+				Amount:     tc.amt,
+				TargetConf: 6,
+			}
+
+			// If we don't expect the send coin to work, we should
+			// simply check for an error.
+			if tc.sendCoinShouldFail {
+				alice.RPC.SendCoinsAssertErr(sendCoinReq)
+				return
+			}
+
+			alice.RPC.SendCoins(sendCoinReq)
+			block := ht.MineBlocksAndAssertNumTxes(
+				1, 1,
+			)[0]
+			tx := block.Transactions[1]
+
+			if len(selectedOutpoints) > 0 {
+				require.Equal(
+					t, len(tx.TxIn), len(selectedOutpoints),
+				)
+
+				for _, in := range tx.TxIn {
+					Op := in.PreviousOutPoint.Hash.String()
+					_, ok := selectedOpLookup[Op]
+					require.True(t, ok)
+				}
+			}
+
+			if sendCoinReq.SendAll && tc.selectedCoins == nil {
+				ht.AssertWalletAccountBalance(
+					alice, lnwallet.DefaultAccountName,
+					0, 0,
+				)
+			} else {
+				ht.AssertConfWalletBalanceLT(
```

Comment:
I think we don't need this new method if we can calculate the fees that the `SendCoins` transaction cost. Since we have the outputs of the transaction stored in `tx` we can easily find the difference in value to `totalInitialCoinSum`, and then we can call 
```
ht.AssertWalletAccountBalance(
	alice, lnwallet.DefaultAccountName,
	int64(totalinitialCoinSum)-tc.amt-fees, 0,
)
```

Does that sound right?

Reply:
Looks good now.